### PR TITLE
Json logging option

### PR DESCRIPTION
**Pull request by Astaelan**
This is only the bare minimum, it creates a json object and plants the existing output into the message field, and removes the color coding.
This can be evolved to include more fields into the json object as appropriate, but will require a bit more tuning of the logging system to enable adding keys from ks_log calls, best left for another issue as this satisfies the base requirements.

### CODE REVIEW

File: src/ks_log.c [hunk 1]
```diff
@@ -307,19 +260,26 @@ static void default_logger(const char *file, const char *func, int line, int lev
...
-	// Add to the buffer after the color, save space for color reset
-	len = ks_log_format_output(buf + prefixlen, sizeof(buf) - prefixlen - suffixlen, file, func, line, level, fmt, ap);
+	len = ks_log_format_output(buf, sizeof(buf), file, func, line, level, fmt, ap);
 
 	if (len > 0) {
+		if (ks_log_jsonified) {
+			ks_json_t *json = ks_json_create_object();
+			ks_json_add_string_to_object(json, "message", buf);
+			char *tmp = ks_json_print_unformatted(json);
+			strncpy(buf, tmp, sizeof(buf) - 2); // safe truncation if neccessary, leaving 2 bytes for newline and terminator
```
Comment (crienzo) [inline src/ks_log.c:270]: You could just fwrite tmp then fwrite ('\n') instead.  One less copy and no chance of truncating the JSON log.

File: src/ks_log.c [hunk 2]
```diff
@@ -307,19 +260,26 @@ static void default_logger(const char *file, const char *func, int line, int lev
...
+			buf[sizeof(buf)] = 0; // gaurentee temporary termination if strncpy used all the data
+			len = strlen(buf); // reassign len, which is used for both console and file output length
+			buf[len] = '\n'; // add a newline to each json output
+			buf[len + 1] = '\0'; // fix the null term we just replaced
+			++len; // add the new line character to the output length
+			free(tmp); // cleanup
+			ks_json_delete(&json);
+		}
+
 		ks_spinlock_acquire(&g_log_spin_lock);
```
Comment (crienzo) [inline src/ks_log.c:280]: replace with mutex

File: src/ks_log.c (hunk 1 above)
Comment (crienzo) [reply src/ks_log.c:270]: fprintf() probably an even better choice for JSON output

Comment (Astaelan) [reply src/ks_log.c:270]: I was trying to avoid restructuring all the logging code, but we can go that way.

File: tests/testwebsock2.c [hunk 3]
```diff
@@ -306,7 +306,7 @@ static int test_ws(char *url)
 	ks_json_t *req = ks_json_create_object();
 	ks_json_add_string_to_object(req, "url", url);
 
-	ks_global_set_default_logger(7);
+	ks_global_set_log_level(7);
```
Comment (crienzo) [inline tests/testwebsock2.c:309]: enable json logging on this test

**Review by crienzo — APPROVED** [review]

**Pull request merged**